### PR TITLE
expand spec sections on JTA and Jakarta Interceptors

### PR DESCRIPTION
**Pull request by gavinking**
I'm trying to make very clear who is responsible for what, and which things are prescriptive requirements on the implementation of Jakarta Data, vs just being information about how Jakarta Data can take advantage of CDI/JTA/Interceptors.

### CODE REVIEW

**Comment by gavinking** [comment]
This is for #473, by the way.

File: spec/src/main/asciidoc/jakarta-ee.adoc [hunk 1]
```diff
@@ -107,16 +107,42 @@ NOTE: CDI Lite (corresponding to Jakarta Core profile) does not include a requir
...
+1. Jakarta Transactions is available,
+2. a global transaction is active on the thread of execution in which a repository operation is called, and
+3. the data source backing the repository is capable of transaction enlistment,
+
+then the repository operation must be performed within the context of the global transaction. That is, the data source resources involved in the operation must be enlisted as participants in the transaction.
+
+[NOTE]
+====
+In the Jakarta EE environment, such enlistment usually happens automatically when the repository implementation makes use of a data source which is under the management of the Jakarta EE container.
+In the case of a data source which is _not_ managed by the container, the repository implementation itself is required to ensure that such enlistment occurs.
```
Comment (njr-11) [inline spec/src/main/asciidoc/jakarta-ee.adoc:134]: Data sources that are not manged by the container fall outside of Jakarta EE so we should avoid requirements in that area.
```suggestion
```

**Review by njr-11 — COMMENTED** [review]
This looks great. Thanks for making this section better. I did spot one requirement that needs to be omitted because it can be interpreted to imply that data sources not managed by the container need to be supported in Jakarta EE products, which goes beyond Jakarta EE.

File: spec/src/main/asciidoc/jakarta-ee.adoc (hunk 1 above)
Comment (gavinking) [reply spec/src/main/asciidoc/jakarta-ee.adoc]: OK.

**Review by njr-11 — APPROVED** [review]

**Pull request merged**